lib/app-components/addon/components/branded-navbar/component.ts

```diff
@@ -50,7 +50,7 @@ export default class BrandedNavbar extends Component {
     }
 
     get submitPreprintUrl() {
-        return this.theme.isProvider ? `${osfURL}preprints/${this.theme.id}/submit/` : `${osfURL}preprints/submit/`;
+        return `${this.theme.pathPrefix}/submit/`;
```

Comment:
There is a comment in `branded-navbar/template.hbs` to make the `OsfLink` that uses this property do an in-app transition (use `@route` and `@models` instead of `@href`). Can you address that comment in this PR and I think we can get rid of this getter altogether? Let me know if you have questions on this too